src/java/org/apache/cassandra/io/util/RandomAccessReader.java

```diff
@@ -479,4 +482,17 @@ public static RandomAccessReader open(File file)
             throw t;
         }
     }
+
+    private ByteBuffer getTemporaryBuffer(int size)
+    {
+        if (temporaryBuffer != null
+            && temporaryBuffer.capacity() == size)
+        {
+            temporaryBuffer.clear();
+            return temporaryBuffer;
+        }
+        temporaryBuffer = ByteBuffer.allocateDirect(size);
```

Comment:
Is it intentional that you replaced `allocate` with `allocateDirect`?

Reply:
This was intentional, on a second thought I am changing this to create a buffer of the same type of the backing buffer.